docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+# ADR-102: RPC Companion
+
+## Changelog
+
+- 2022-03-27: First draft (@andynog)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+This ADR proposes an architecture of an ***RPC Companion*** solution implemented based on the proposed [ADR-101 Data Companion Pull API](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md). This solution can run as a sidecar concurrently with the full node, and it is optional.
```

Comment:
A sidecar here is a separate process?

Reply:
Yes. That should probably be emphasized here.